src/Hangfire.Core/BackgroundJobServer.cs

```diff
@@ -163,5 +174,47 @@ public void Start()
         public void Stop()
         {
         }
+
+        public void OnNext(int i)
+        {
+            int threshold = int.Parse(ConfigurationManager.AppSettings["failedJobCountThreshold"]);
+            int delay = int.Parse(ConfigurationManager.AppSettings["failedJobCheckInterval"]);
+            TimeSpan duration = new TimeSpan(0, delay, 0);
+
+            lock (_failedLock)
+            {
+                _failedJobsCount++;
+            }            
+
+            if (!_hasFirstFailedDateTime || DateTime.Now.Subtract(duration) > _lastEmailNotification)
+            {
+                _lastEmailNotification = DateTime.Now;
+                _failedJobsCount = 1;
+                _hasFirstFailedDateTime = true;
+            }
+            else if(_failedJobsCount > threshold )
+            {
+                string msg = "The server with machine name: \n";
+                msg += '\n' + Environment.MachineName + ", ";
+                msg += "\n \n has had a peak of failed jobs.";
+
+                ServerStatusNotifier.Notify(1, msg);
+                _hasFirstFailedDateTime = false;
+            }
+        }
+
+        public void OnError(Exception error)
+        {
+            throw new NotImplementedException();
+        }
+
+        public void OnCompleted()
+        {
+            throw new NotImplementedException();
```

Comment:
you sure that this will not just throw a random exception? also same with the one **above**.